### ISSUE TITLE
Getter event

### ISSUE THREAD

**Issue by krzysztofantczak** [ElliotNB/observable-slim#11]
Hey,

First of all, i love how i can get info about add/update on object properties, but.... Is it possible to somehow get notified when an object property is used? Like simple:

```
console.log(obj.some_prop);
```

ObservableSlim callback would be called with "read" type, or something like that? Even ugly and hacky way is good enough for my use case.

**Comment by ElliotNB** [comment]
Currently Observable Slim doesn't support this, but it's a great enhancement idea. I'm going to look into this one and hopefully have the functionality added within a few days. 

We have to be very careful with how we modify the `get` handler in the library. The `get` handler is invoked whenever a property is accessed -- iterating over large nested objects or merging objects will invoke the `get` handler many many times. It's important that the `get` handler remain as highly performant as possible in order to ensure the library doesn't bog down with large nested objects.

**Comment by krzysztofantczak** [comment]
You are right. For my use case it's enough, when last prop gets the event. So for example, if i'll be reading path foo.bar.baz - baz will be the accessed property of object bar - no need to get another event on bar as property of foo. Well, unless i will read foo.bar :D Not sure if that's doable this way.

**Comment by ElliotNB** [comment]
Unfortunately it's not possible to only trigger the `get` handler for only a specific set of properties -- accessing any of the properties on an observed object would trigger the `get` handler.

In your example above, accessing the value `foo.bar.baz` would trigger the `get` handler twice -- once for `foo.bar` and another time for `bar.baz`. If you performed something like a object deep merge (e.g., the jQuery `$.extend(true, {}, {})` the number of times the `get` handler is invoked can skyrocket if the objects are large and deeply nested. Keeping the code highly performant becomes very important -- particularly for browsers that requires the `Proxy` polyfill like IE11.
